zebra-state/src/service/finalized_state.rs

```diff
@@ -390,6 +396,15 @@ impl FinalizedState {
                 batch.zs_insert(history_tree_cf, height, history_tree);
             }
 
+            let current_pool = self.current_value_pool().constrain::<NegativeAllowed>()?;
+            batch.zs_insert(
+                tip_chain_value_pool,
+                height,
+                current_pool
+                    .update_with_chain_value_pool_change(block_value_balance)?
+                    .constrain::<NonNegative>()?,
```

Comment:
Do we need to constrain here?
```suggestion
                    .update_with_chain_value_pool_change(block_value_balance)?,
```